src/pages/components/map-related/useRecenterOnDataChange.ts

```diff
@@ -3,17 +3,31 @@ import { LatLngTuple } from 'leaflet'
 import { useMap } from 'react-leaflet'
 import { MapProps } from './map-types'
 
-export function useRecenterOnDataChange({ positions }: MapProps) {
+export function useRecenterOnDataChange({ positions = [], plannedRouteStops = [] }: MapProps) {
   const map = useMap()
-  const positionsSum = positions.reduce(
-    (acc, { loc }) => [acc[0] + loc[0], acc[1] + loc[1]],
-    [0, 0],
-  )
-  const mean: LatLngTuple = [positionsSum[0] / positions.length, positionsSum[1] / positions.length]
-  console.log('mean: ', mean)
+
   useEffect(() => {
+    if (!positions.length && !plannedRouteStops.length) return
+
+    const positionsSum = positions.reduce(
+      ([lat, lng], pos) => [lat + pos.loc[0], lng + pos.loc[1]],
+      [0, 0],
+    )
+    const stopsSum = plannedRouteStops.reduce(
+      ([lat, lng], stop) => [lat + stop.location.latitude, lng + stop.location.longitude],
+      [0, 0],
+    )
+
+    const totalPoints = positions.length + plannedRouteStops.length
+
+    const mean: LatLngTuple = [
+      (positionsSum[0] + stopsSum[0]) / totalPoints,
+      (positionsSum[1] + stopsSum[1]) / totalPoints,
+    ]
+
+    // console.log('mean: ', mean)
     if (mean[0] || mean[1]) {
       map.setView(mean, map.getZoom(), { animate: true })
     }
-  }, [mean])
+  }, [positions, plannedRouteStops, map])
```

Comment:
I'm not sure we want to update the view of the map every time the reference to the positions array is changing.
I think that calculating the mean is a cheap operation and that's why I placed it outside `useEffect`, I didn't care it that will happen every render, but moving the map too often can affect the user experience